app/controllers/api/v2/hostgroups_controller.rb

```diff
@@ -73,6 +73,9 @@ def create
         @hostgroup.suggest_default_pxe_loader if params[:hostgroup] && params[:hostgroup][:pxe_loader].nil?
 
         process_response @hostgroup.save
+      rescue ActiveRecord::InvalidForeignKey => error
```

Comment:
Could this be put to a more generic place? the code looks universal but addresses the issue only for one host group action. We do something similar in UI for uknown records https://github.com/theforeman/foreman/blob/develop/app/controllers/application_controller.rb#L11 and API https://github.com/theforeman/foreman/blob/develop/app/controllers/api/base_controller.rb#L37-L45